### PR TITLE
Release automation for docs

### PR DESCRIPTION
**Pull request by willdavsmith**
Thank you for helping make the Radius documentation better!

**Please follow this checklist before submitting:**

- [ ] [Read the contribution guide](https://docs.radapp.dev/contributing/docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

### Summary
🚀📝🧪

This pull request adds scripts and a workflow to automate the release process of the docs website for different versions of Radius. It allows the user to specify a version number and creates a corresponding release branch and default branch for the docs repository. It also validates the version number and updates the configuration and content files accordingly.

> _`release-docs.sh`_
> _Automates website updates_
> _A winter blossom_

### Walkthrough
*  Add a GitHub workflow to manually release the docs website for a given version of Radius ([link](https://github.com/project-radius/docs/pull/696/files?diff=unified&w=0#diff-e426ed45842837026e10e66af23d9c7077e89eacbe6958ce7cb991130ad05adaR1-R44))
  - Validate the version number with the `validate_semver.py` script ([link](https://github.com/project-radius/docs/pull/696/files?diff=unified&w=0#diff-3dd4e2666be1d9e3b86df87bb30d9a6a159aa00e000bd42becba94cd4dc95dcfR1-R45))
* Add a script to automate the release process of the docs website ([link](https://github.com/project-radius/docs/pull/696/files?diff=unified&w=0#diff-8d44f3e5840dec47347183ed9abd79878fd705bf81dc761cf03cada198e2b555R1-R75))
* Add a script to validate the version number format ([link](https://github.com/project-radius/docs/pull/696/files?diff=unified&w=0#diff-3dd4e2666be1d9e3b86df87bb30d9a6a159aa00e000bd42becba94cd4dc95dcfR1-R45))

### CODE REVIEW

**Review by youngbupark — DISMISSED** [review]
lgtm but please the review from @AaronCrawfis :)

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        run: |
+          # CHANNEL is the major and minor version of the VERSION_NUMBER (e.g. 0.1)
+          CHANNEL="$(echo ${{ inputs.version }} | cut -d '.' -f 1,2)"
+          echo "channel=$CHANNEL" >> $GITHUB_OUTPUT
+      - name: Release docs
+        run: |
+          ./docs/.github/scripts/release-docs.sh ${{ inputs.version }}
+      - name: Change the default branch
+        run: |
+          gh api \
```
Comment (AaronCrawfis) [inline .github/workflows/release.yaml:44]: Will the default Actions token have the ability to change the default branch, or will we have to use GH_RAD_CI_BOT_PAT?

File: .github/scripts/release-docs.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+# In docs/config.toml, change version to VERSION instead of edge
+VERSION_STRING_REPLACEMENT="version = \"${CHANNEL_VERSION}\""
+awk -v REPLACEMENT="${VERSION_STRING_REPLACEMENT}" '{gsub(/version = \"edge\"/, REPLACEMENT); print}' docs/config.toml > docs/config.toml.tmp
+mv docs/config.toml.tmp docs/config.toml
+
+# In docs/config.toml, change chart_version (Helm chart) to VERSION_NUMBER
+CHART_VERSION_STRING_REPLACEMENT="chart_version = \"${VERSION_NUMBER}\""
+awk -v REPLACEMENT="${CHART_VERSION_STRING_REPLACEMENT}" '{gsub(/chart_version = \"[^\n]+\"/, REPLACEMENT); print}' docs/config.toml > docs/config.toml.tmp
+mv docs/config.toml.tmp docs/config.toml
+
```
Comment (AaronCrawfis) [inline .github/scripts/release-docs.sh:65]: Can we update `github_branch` to the new branch name as well please?

File: .github/workflows/release.yaml (hunk 1 above)
Comment (willdavsmith) [reply .github/workflows/release.yaml:44]: good call, fixed to add this to the env

File: .github/scripts/release-docs.sh (hunk 2 above)
Comment (willdavsmith) [reply .github/scripts/release-docs.sh:65]: updated

**Review by AaronCrawfis — DISMISSED** [review]

**Review by AaronCrawfis — APPROVED** [review]

**Pull request merged**